src/partykit/index.ts

```diff
@@ -49,3 +49,41 @@ export const connect: ConnectFunction = (
     return connection;
   });
 };
+
+async function getOrCreateRemoteName(dbName: string) {
+  const petnames = fireproof("petname.mappings");
+
+  try {
+    const doc = await petnames.get<{ remoteName: string; firstConnect: boolean }>(dbName);
+    return { remoteName: doc.remoteName, firstConnect: false };
+  } catch (_error) {
+    const remoteName = petnames.sthis.nextId().str;
+    await petnames.put({ _id: dbName, remoteName, firstConnect: true });
+    return { remoteName };
+  }
+}
+
+export function cloudConnect(
+  db: Database,
+  dashboardURI = BuildURI.from("https://dashboard.fireproof.storage/"),
```

Comment:
The BuildURI is a kind of mutable better use URI to express that you don't expect that the URI is modified after you passed it in. Or other said do not pass around mutable Objects.

Reply:
when I try that I get 'Property 'build' does not exist on type 'BuildURI''